pyvisa_sim/testsuite/test_common.py

```diff
@@ -6,35 +6,72 @@
 
 
 @pytest.mark.parametrize(
-    "data, mask, send_end, want",
+    "bits, want",
     [
-        (b"1234", None, False, b"1234"),
-        (b"41234", 3, False, b"40004"),
-        # Can't figure this one out. Getting ValueError: bytes must in in range(0, 256)
-        # If I knew more about the purpose of `iter_bytes` then maybe I could
-        # reconcile it, but I don't have time to investigate right now.
-        pytest.param(
-            b"1234",
-            3,
-            True,
-            b"1234",
-            marks=pytest.mark.xfail(
-                reason=(
-                    "ValueError bytes must be in range(0, 256). TODO: figure"
-                    " out correct 'want' value."
-                )
-            ),
-        ),
+        (0, 0b0),
+        (1, 0b1),
+        (5, 0b0001_1111),
+        (7, 0b0111_1111),
+        (8, 0b1111_1111),
+        (11, 0b0111_1111_1111),
+    ],
+)
+def test_create_bitmask(bits, want):
+    got = common._create_bitmask(bits)
+    assert got == want
+
+
+@pytest.mark.parametrize(
+    "data, data_bits, send_end, want",
+    [
+        (b"\x01", None, False, b"\x01"),
+        (b"hello world!", None, False, b"hello world!"),
+        # Only apply the mask
+        (b"\x03", 2, None, b"\x03"),  # 0b0000_0011 --> 0b0000_0011
+        (b"\x04", 2, None, b"\x00"),  # 0b0000_0100 --> 0b0000_0000
+        (b"\xff", 5, None, b"\x1f"),  # 0b1111_1111 --> 0b0001_1111
+        (b"\xfe", 7, None, b"\x7e"),  # 0b1111_1110 --> 0b0111_1110
+        (b"\xfe", 8, None, b"\xfe"),  # 0b1111_1110 --> 0b1111_1110
+        (b"\xff", 9, None, b"\xff"),  # 0b1111_1111 --> 0b1111_1111
+        # Always set highest bit *of data_bits* to 0
+        (b"\x04", 2, False, b"\x00"),  # 0b0000_0100 --> 0b0000_0000
+        (b"\x04", 3, False, b"\x00"),  # 0b0000_0100 --> 0b0000_0000
+        (b"\x05", 3, False, b"\x01"),  # 0b0000_0101 --> 0b0000_0001
+        (b"\xff", 7, False, b"\x3f"),  # 0b1111_1111 --> 0b0011_1111
+        (b"\xff", 8, False, b"\x7f"),  # 0b1111_1111 --> 0b0111_1111
```

Comment:
Those cases need to cover more than a single byte.

Reply:
See L54-L59 - those cases test multiple bytes.

Reply:
Yes and they do expose the issue about clearing the highest bit I mentioned in another comment.